src/navigationDrawer/navigationDrawer.component.tsx

```diff
@@ -77,11 +37,14 @@ const ForwardRefLink = React.forwardRef<HTMLAnchorElement, LinkProps>(
 );
 ForwardRefLink.displayName = 'ForwardRefLink';
 
-class NavigationDrawer extends Component<CombinedNavigationProps> {
-  private createLink(plugin: PluginConfig, index: number): React.ReactElement {
-    const imgSrc = this.props.darkMode
-      ? plugin.logoDarkMode
-      : plugin.logoLightMode;
+export const NavigationDrawer = (
+  props: NavigationDrawerProps
+): React.ReactElement => {
+  const createLink = (
```

Comment:
I see you've migrated this component from a class component to a functional component - however we should avoid defining functions within function components. Can you either extract them to be defined outside of the `NavigationDrawer` component or wrap these functions in `useCallback`? As otherwise these functions are recreated on every render of `NavigationDrawer`

Reply:
Yes I ended up doing this to the tour as well during the initial upgrade as I had to move everything that used `connect` to use `makeStyles` instead of `withStyles` in order to get it to run before I refactored all the styles. I can technically go back to the class components now but I will try keep the functional one if I can since that's the newer method.

Reply:
I'm happy with the class -> functional component migration, I was just highlighting that something that used to be fine with class components (functions defined within the component) aren't efficient in functional ones